R/carehomes_rt.R

```diff
@@ -224,6 +224,199 @@ carehomes_Rt <- function(step, S, p, prob_strain = NULL,
+##'
+##' @param save_all A boolean determining whether to save all samples of Rt
+##' estimated or only a summary
+##'
+##' @return A list with elements
+##' `t_start` (vector of first days of the sliding windows over which Rt is
+##' estimated),
+##' `t_end` (vector of last days of the sliding windows over which Rt is
+##' estimated),
+##' `Rt` a matrix (only present if `save_all = TRUE`) containing for each
+##' sliding window (each row in the matrix) a sample of n_R * nrow(inc) values
+##' of Rt for that sliding window (columns of the matrix)
+##' `Rt_summary` a matrix containing for each sliding window (each row in the
+##' matrix) the 2.5%, 50%, 97.5% quantiles and the mean of Rt for that sliding
+##' window (columns of the matrix)
+##'
+##' @export
+##' @importFrom EpiEstim make_config estimate_R
+##' @importFrom epitrix gamma_mucv2shapescale
+##' @importFrom stats quantile rgamma
+carehomes_EpiEstim_Rt_trajectories <- function(step, incidence, p,
+                                  sliding_window_ndays = 7,
+                                  mean_prior = 1,
+                                  sd_prior = 1,
+                                  n_GT = 10000,
+                                  n_R = 1000,
+                                  save_all = TRUE) {
+  gt_distr <- draw_one_GT_distr(p = p, n = n_GT)
+
+  T <- ncol(incidence)
+  np <- nrow(incidence)
+  t_start <- seq(2, T - sliding_window_ndays + 1)
+  t_end <- seq(sliding_window_ndays + 1, T)
+
+  # generate the config based on incidence for first parameter set
+  config <- EpiEstim::make_config(incid = incidence[1, ],
+                                  t_start = t_start,
+                                  t_end = t_end,
+                                  method = "non_parametric_SI",
+                                  si_distr = gt_distr,
+                                  mean_prior = mean_prior,
+                                  std_prior = sd_prior)
+
+  R_sample <- matrix(NA, length(t_start), n_R * np)
+
+  for (i in seq_len(np)) {
+    ## this function gives warning when precision in estimates is not good
+    ## but we don't care
+    R_i <- suppressWarnings(EpiEstim::estimate_R(incid = incidence[i, ],
+                                                 config = config)$R)
+
+    R_i_shape_scale <-
+      epitrix::gamma_mucv2shapescale(mu = R_i$`Mean(R)`,
+                                     cv = R_i$`Std(R)` / R_i$`Mean(R)`)
+
+    R_sample[, n_R * (i - 1) + seq_len(n_R)] <-
+      t(sapply(seq_len(length(t_start)), function(e) {
+        if (!is.na(R_i_shape_scale$shape[e])) {
+          ret <- rgamma(n_R, shape = R_i_shape_scale$shape[e],
+                        scale = R_i_shape_scale$scale[e])
+        } else {
+          ret <- rep(NA, n_R)
+        }
+        ret
+      }))
+  }
+
+  summary_R <- apply(R_sample, 1, quantile, c(0.025, 0.5, 0.975), na.rm = TRUE)
+  mean_R <- apply(R_sample, 1, mean, na.rm = TRUE)
+  summary_R <- t(rbind(summary_R, mean_R))
```

Comment:
I am currently in the process of refactoring the Rt methods to return time in the last dimension to match the rest of the functions in the package. Could you make sure yours do that too from the start